app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/login/LoginViewModel.kt

```diff
@@ -0,0 +1,88 @@
+package com.islaharper.dawnofandroid.presentation.screens.login
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.islaharper.dawnofandroid.domain.model.ApiResponse
+import com.islaharper.dawnofandroid.domain.model.ApiTokenRequest
+import com.islaharper.dawnofandroid.domain.model.MessageBarState
+import com.islaharper.dawnofandroid.domain.useCases.readSignedInState.ReadSignedInStateUseCase
+import com.islaharper.dawnofandroid.domain.useCases.saveSignedInState.SaveSignedInStateUseCase
+import com.islaharper.dawnofandroid.domain.useCases.verifyToken.VerifyTokenUseCase
+import com.islaharper.dawnofandroid.util.Resource
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import java.net.ConnectException
+import java.net.SocketTimeoutException
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val readSignedInStateUseCase: ReadSignedInStateUseCase,
+    private val saveSignedInStateUseCase: SaveSignedInStateUseCase,
+    private val verifyTokenUseCase: VerifyTokenUseCase,
+) : ViewModel() {
+    private val _signedInState: MutableState<Boolean> = mutableStateOf(false)
```

Comment:
I don't think these properties should be exposed as `androidx.compose.runtime.MutableState`, it should be a `StateFlow`. If I am not mistaken, according to the Google official documentation the only time we should favor `MutableState` is where we are using `TextField` where the internal implementations of the Compose text fields make it extremely tricky to use `Flow`

Reply:
Refactored and in latest commit.